client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -67,6 +68,17 @@
     CompletableFuture<Boolean> updateStream(final StreamConfiguration streamConfig);
 
     /**
+     * Api to update stream.
+     *
+     * @param scope      Scope
+     * @param streamName Stream
+     * @param streamCut  Stream cut to updated
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     * indicate that the stream was truncated at the supplied cut.
+     */
+    CompletableFuture<Boolean> truncateStream(final String scope, final String streamName, final Map<Integer, Long> streamCut);
```

Comment:
yes, but stream cut that is currently exposed to application has private access to the map of segment to offset as that information is opaque to application.

So i have avoided creating a new data structure here because if readergroup has access to map inside the stream cut then it can extract that and pass it to controller client directly.

Reply:
StreamCut's position map is not exposed outside the package. 
This means we cant use that object in controller interface's implementation outside this package (e.g. local controller etc).

Reply:
I have exposed two apis on controller now.. truncate(Map<Int,Long>) and truncate(StreamCut)

Both apis are implement on controllerImpl
However, only former is implemented on LocalController

Reply:
If you want to resolve this in this pull request so that we make it cleaner, then fine. `StreamCut` should be public with the direction we are going.

Reply:
Stream cut is public. Only its internal details are hidden from the application. (the positions map). 
And IMO we should continue to keep it opaque to the application.